### ISSUE TITLE
`LatitudeLongitudeGrid` may have incorrect metrics for `y`-Periodic domains

### ISSUE THREAD

**Issue by glwagner** [CliMA/Oceananigans.jl#3465]
This code:

```Julia
using Oceananigans
grid = LatitudeLongitudeGrid(longitude = (215, 215.25),
                             latitude = (50, 50.25),
                             size = (1, 1, 50),
                             halo = (1, 1, 1),
                             z = (-200, 0),
                             topology = (Periodic, Periodic, Bounded))
```

produces a `BoundsError`, if run with `julia --check-bounds=yes`.

The problem is the worksize specified here:

https://github.com/CliMA/Oceananigans.jl/blob/3b02321a350003fa9596c4f40839b031a5cc7ad1/src/Grids/latitude_longitude_grid.jl#L443

I also don't like this code because we cannot read it to understand it. For example, what is `length(grid.φᵃᶜᵃ)`? Is this `Ny + Hy`, or `Ny + Hy + 1`? I think it would be better to be explicit.

It appears that for both `Periodic` and `Bounded` in `y`, then we have

```julia
length(grid.φᵃᶜᵃ) = 3
```

Which looks like `Ny + 2Hy`.

I think both the worksize and the metric size is wrong in general --- I think we have hardcoded a `Bounded` y-topology. I think the right thing to do is to go in and use `total_length` to determine both the correct size of the metrics for any topology, as well as the size of the kernel that is launched to compute them.

**Comment by glwagner** [comment]
A quick fix is to use

```julia
 @inline metric_worksize(grid::XRegularLLG)  = length(grid.φᵃᶠᵃ) - 2
```

**Comment by navidcy** [comment]
What's the `metric_worksize`?

**Comment by glwagner** [comment]
I guess it's the size of the kernel that we use to compute the metrics. Basically it has to be the number of faces - 2, beacause we compute metrics using differences.

**Comment by glwagner** [comment]
Closed by #3450

**Comment by navidcy** [comment]
> Closed by #3450

since #3450 includes other things, perhaps you can point out the source code change or the commit the resolved the issue?
